src/search/Decoder.ts

```diff
@@ -16,36 +16,24 @@
  * limitations under the License.
  */
 
-import { AbstractTestCase } from "../AbstractTestCase";
-import { Statement } from "../statements/Statement";
+import {Encoding} from "./Encoding";
 
 /**
- * TestCaseDecoder interface.
+ * Decoder interface.
  *
  * @author Dimitri Stallenberg
  * @author Mitchell Olsthoorn
  */
-export interface TestCaseDecoder {
+export interface Decoder<E extends Encoding, D> {
```

Comment:
Add comment about what the generics mean

Reply:
Done